server/logger.js

```diff
@@ -0,0 +1,100 @@
+import winston from 'winston';
+import fs from 'fs';
+import { fileURLToPath } from 'url';
+import path from 'path';
+
+const { combine, timestamp, printf, colorize } = winston.format;
+
+const __filename = fileURLToPath(import.meta.url);
+const __dirname = path.dirname(__filename);
+
+const ROOT = __dirname;
```

Comment:
In the `removeSocket` function, you are correctly deleting the user token from both `sockets` and `socket_user_map`. However, it would be beneficial to add error handling to ensure that the `usertoken` exists before attempting to delete it. This can prevent potential runtime errors. 

```diff
+   if (!sockets[usertoken]) {
+       throw new Error('usertoken does not exist');
+   }
```


    *This comment was auto-generated by **Code Bot** to provide suggestions and improvements.*